app/tiebout.hoon

```diff
@@ -0,0 +1,253 @@
+::
+::::  /app/tiebout/hoon
+::
+/?  309
+/-  hall
+/+  tiebout
+=,  tiebout
+=,  eyre
+::
+::  state:
+::
+~%  %tiebout  ..^is  ~
+|_  [bol=bowl:gall sta=state]
+::
+::  +this: app core subject
+::
+++  this  .
+::
+::  +prep: set up app state, upgrade app state
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    :-  ~
+    %=  this
+      king.tie.sta  ~dopzod
+      baseurl.tie.sta  'https://api.push.apple.com/3/device/'
+    ==
+  ?-  -.u.old
+    %0
+      [~ this(sta u.old)]
+  ==
+::
+::  +coup: receive acknowledgement for poke, print error if it failed
+::
+++  coup
+  |=  [wir=wire err=(unit tang)]
+  ^-  (quip move _this)
+  ?~  err
+    [~ this]
+  (mean u.err)
+::
+::  +poke-noun: receive debugging actions
+::
+++  poke-noun
+  |=  act=action
+  ^-  (quip move _this)
+  (poke-tiebout-action act)
+::
+::  +poke-tiebout-action: main action handler
+::
+++  poke-tiebout-action
+  |=  act=action
+  ^-  (quip move _this)
+  ?-  -.act
+    $king
+      (set-king +.act)
+    $token
+      (set-token +.act)
+    $baseurl
+      (set-baseurl +.act)
+    $add-circle
+      (add-circle +.act)
+    $del-circle
+      (del-circle +.act)
+    $notify
+      (send-notify +.act)
+  ==
+::
+::  +add-circle: add circle and subscribe for updates
+::
+++  add-circle
+  |=  nom=name:hall
+  ^-  (quip move _this)
+  :_  this(circles.tie.sta (~(put by circles.tie.sta) nom 0))
+  [ost.bol %peer /our/[nom] [our.bol %hall] /circle/[nom]/config/grams]~
```

Comment:
There's a lot of move construction happening in separate arms. There's common themes, like `ost.bol` at the head, or `[our.bol %hall]` being used for the `dock`. You may or may not want to write arms that deal with the common parts of move construction for you.